haystack/components/retrievers/sentence_window_retriever.py

```diff
@@ -304,7 +304,7 @@ async def _retrieve_context_for_document_async(self, doc: Document, window_size:
 
         assert split_id is not None
         filter_conditions = self._build_filter_conditions(split_id, window_size, source_ids)
-        context_docs = await self.document_store.filter_documents_async(filter_conditions)
+        context_docs = await self.document_store.filter_documents_async(filter_conditions)  # type: ignore
```

Comment:
Could you specifically provide the label we are ignoring e.g. for `misc` it would look like `# type: ignore[misc]`?

Also leave a dev comment explaining why we are ignoring